docs/pipelines.md

```diff
@@ -226,6 +227,58 @@ spec:
           value: /workspace/examples/microservices/leeroy-web
 ```
 
+### Tekton Bundles
+
+You may also specify your `Task`
+reference using a `Tekton Bundle`. A `Tekton Bundle` is an OCI artifact that
+contains Tekton resources like `Tasks` and `Pipelines` which can be referenced
+within a `taskRef`.
+
+ ```yaml
+ spec:
+   tasks:
+     - name: hello-world
+       taskRef:
+         name: echo-task
+         bundle: docker.com/myrepo/mycatalog
+ ```
+
+Here, the `bundle` field is the full reference url to the artifact. The name is the
+`metadata.name` field of the `Task`. 
+
+You may also specify a `tag` as you would with a Docker image which will give you a fixed,
+repeatable reference to a `Task`.
+
+ ```yaml
+ spec:
+   tasks:
+   - name: hello-world
+     taskRef:
+       name: echo-task
+       bundle: docker.com/myrepo/mycatalog:v1.0.1
+ ```
+
+You may also specify a fixed digest instead of a tag.
+
+ ```yaml
+ spec:
+   tasks:
+   - name: hello-world  
+     taskRef:
+       name: echo-task
+       bundle: registry.hub.docker.com/myrepo/mycatalog@sha256:abc123
```

Comment:
we should probably use a well formed (real) digest here, maybe even a real image, like `busybox`.

Reply:
Would that possible confuse a user to use a real image since we can't actually use a busybox image since it doesn't conform to the contract. I was trying to make it obvious here that this is an example which is why the digest isn't a real one. Do you think there is value in making this a real example?

Reply:
@pierretasci my bad :facepalm: Indeed we shouldn't use an image that doesn't satisfy the contract.